src/messages/MMgrConfigure.h

```diff
@@ -65,9 +62,20 @@ class MMgrConfigure : public MessageInstance<MMgrConfigure> {
   }
 
   MMgrConfigure()
-    : MessageInstance(MSG_MGR_CONFIGURE, HEAD_VERSION, COMPAT_VERSION)
+    : Message{MSG_MGR_CONFIGURE, HEAD_VERSION, COMPAT_VERSION}
   {}
+
+private:
+  using RefCountedObject::put;
+  using RefCountedObject::get;
+  friend void intrusive_ptr_add_ref(const MMgrConfigure* m) {
+    m->get();
+  }
+  friend void intrusive_ptr_release(const MMgrConfigure* m) {
+    m->put();
+  }
+  template<class T, typename... Args>
+  friend boost::intrusive_ptr<T> ceph::make_message(Args&&... args);
```

Comment:
Ah, I was going to leave a comment in `MMgrDigest` if we could specify `make_message` is a friend, but I now see you can in fact do that here. I would advocate marking any message that we convert over to make the constructors private. This way only `make_message` can be used to construct the message.

Reply:
privated.